src/receive_imf.rs

```diff
@@ -2280,48 +2345,24 @@ async fn apply_group_changes(
         let mut added_ids = HashSet::<ContactId>::new();
         let mut removed_ids = HashSet::<ContactId>::new();
 
-        if !recreate_member_list {
-            if sync_member_list {
+        if !self_added {
+            if mime_parser.get_header(HeaderDef::ChatVersion).is_none() {
+                // Allow non-Delta Chat MUAs to add members.
                 added_ids = new_members.difference(&chat_contacts).copied().collect();
-            } else if let Some(added_id) = added_id {
+            }
+
+            if let Some(added_id) = added_id {
                 added_ids.insert(added_id);
             }
+
             new_members.clone_from(&chat_contacts);
             // Don't delete any members locally, but instead add absent ones to provide group
             // membership consistency for all members:
-            // - Classical MUA users usually don't intend to remove users from an email thread, so
-            //   if they removed a recipient then it was probably by accident.
-            // - DC users could miss new member additions and then better to handle this in the same
-            //   way as for classical MUA messages. Moreover, if we remove a member implicitly, they
-            //   will never know that and continue to think they're still here.
-            // But it shouldn't be a big problem if somebody missed a member removal, because they
-            // will likely recreate the member list from the next received message. The problem
-            // occurs only if that "somebody" managed to reply earlier. Really, it's a problem for
-            // big groups with high message rate, but let it be for now.
             new_members.extend(added_ids.clone());
         }
         if let Some(removed_id) = removed_id {
             new_members.remove(&removed_id);
         }
-        if recreate_member_list {
```

Comment:
Why remove this block? It fills `added_ids, removed_ids`, they also should be removed if so. But i'd at least leave `msg_add_member_local()` call for the implicit self addition, i added it recently because the user complained that they continue to receive messages after they left a chat, but there's no any message about the addition back.